File: backend/flowlet/src/main/java/com/example/flowlet/appsetting/domain/model/AppSetting.java
package com.example.flowlet.appsetting.domain.model;

import com.example.flowlet.account.domain.model.PaymentDateAdjustmentRule;

import java.time.LocalDateTime;

public record AppSetting(
    Long appSettingId,
    Integer monthStartDay,
    PaymentDateAdjustmentRule monthStartAdjustmentRule,
    LocalDateTime createdAt,
    LocalDateTime updatedAt
) {
}

File: frontend/src/features/appSetting/types/appSetting.ts
import type { PaymentDateAdjustmentRule } from '../../account/types/account'

export type AppSetting = {
  monthStartDay: number
  monthStartAdjustmentRule: PaymentDateAdjustmentRule
  updatedAt: string
}

export type UpdateAppSettingInput = {
  monthStartDay: number
  monthStartAdjustmentRule: PaymentDateAdjustmentRule
}
